server/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -98,10 +101,17 @@ public XContentBuilder newBuilder(@Nullable XContentType requestContentType, boo
     public XContentBuilder newBuilder(@Nullable XContentType requestContentType, @Nullable XContentType responseContentType,
             boolean useFiltering) throws IOException {
         if (responseContentType == null) {
-            //TODO PG shoudld format vs acceptHeader be always the same, do we allow overriding?
-            responseContentType = XContentType.fromFormat(format);
-            if (responseContentType == null) {
-                responseContentType = XContentType.fromMediaType(acceptHeader);
+            if (Strings.hasText(format)) {
+                responseContentType = XContentType.fromFormat(format);
+            } else if (Strings.hasText(acceptHeader) && acceptHeader.equals("*/*") == false
```

Comment:
since parsing is strict and throws an exception when unkown type is used then it requires a lot of "if guards" to make sure it will not fail.
Also looks to me that `text/plain` is commonly used as returnType for `_cat` api. We should probably add it to xcontentType?

Reply:
not fixing this in current PR. will follow up